### ISSUE TITLE
Endpoint is stuck in waiting-for-identity status

### ISSUE THREAD

**Issue by rlenglet** [cilium/cilium#2232]
- Cilium version:
`Cilium 1.0.0-rc2 40557c1 Sun, 3 Dec 2017 22:10:46 +0100 go version go1.9 linux/amd64`
- Kernel version:
`Linux minikube 4.9.13 #1 SMP Thu Oct 19 17:14:00 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`
- Orchestration system version in use: Kubernetes
`Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.0", GitCommit:"0b9efaeb34a2fc51ff8e4d34ad9bc6375459c4a4", GitTreeState:"clean", BuildDate:"2017-11-29T22:43:34Z", GoVersion:"go1.9.1", Compiler:"gc", Platform:"linux/amd64"}
`

One of my pods (endpoint ID 22252) is stuck in the `waiting-for-identity` status forever.
All other pods started before and after that pod are in the `ready` status.

```
$ kubectl exec cilium-q8wrv -n kube-system -ti cilium endpoint log 22252
Timestamp              Status    State                  Message
2017-12-04T08:35:45Z   OK        waiting-for-identity   Endpoint policy update skipped because no changes were needed
2017-12-04T08:34:48Z   OK        waiting-for-identity   Endpoint policy update skipped because no changes were needed
2017-12-04T08:34:48Z   OK        waiting-for-identity   Endpoint policy update skipped because no changes were needed
2017-12-04T08:34:39Z   OK        waiting-for-identity   Endpoint policy update skipped because no changes were needed
2017-12-04T08:33:57Z   OK        waiting-for-identity   Successfully regenerated endpoint program due to Create endpoint from API PUT
2017-12-04T08:33:57Z   Warning   waiting-for-identity   Skipped invalid state transition to ready due to: Completed endpoint regeneration with no pending regeneration requests
2017-12-04T08:33:33Z   Warning   waiting-for-identity   Skipped invalid state transition to waiting-to-regenerate due to: Create endpoint from API PUT
```
```
$ kubectl logs cilium-q8wrv -n kube-system | grep 22252
time="2017-12-04T08:33:33Z" level=info msg="Invalid state transition skipped" containerID=db0a7f89e2776660bbd7ff48aca81f6fbfe38dff3bdf935d00ffcf6d7cd67059 endpointID=22252 endpointState.from=waiting-for-identity endpointState.to=waiting-to-regenerate file=/tmp/cilium-net-build/src/github.com/cilium/cilium/daemon/endpoint.go line=183 policyRevision=0
time="2017-12-04T08:33:57Z" level=info msg="Regenerated program of endpoint" containerID=db0a7f89e2776660bbd7ff48aca81f6fbfe38dff3bdf935d00ffcf6d7cd67059 endpointID=22252 policyRevision=0
```

**Comment by rlenglet** [comment]
Restarting Cilium fixes the endpoint's status.

**Comment by ianvernon** [comment]
Related: #2227

**Comment by jrajahalme** [comment]
The fix in #2237 is hiding the real problem, which may be that the endpoint does not get an identity at all, like in #2227.

**Comment by tgraf** [comment]
We have never seen this again
